src/SonarScanner.MSBuild.Common/ConsoleLogger.cs

```diff
@@ -124,7 +124,10 @@ public void LogInfo(string message, params object[] args) =>
 
     public void LogUIWarning(string message, params object[] args)
     {
+        var before = IncludeTimestamp; // this is a hack to prevent the timestamp from appearing on the UI, if it was set beforehand.
+        IncludeTimestamp = false;
         uiWarnings.Add(GetFormattedMessage(message, args));
+        IncludeTimestamp = before;
         LogWarning(message, args);
```

Comment:
To avoid saving the state, we can extract a method that does the formatting:
```csharp
    private static string FormatMessage(string message, params object[] args) =>
        args is not null && args.Length > 0
            ? string.Format(CultureInfo.CurrentCulture, message ?? string.Empty, args)
            : message;
```
And this method can be used by the `LogUIWarning` and by the `GetFormattedMessage` methods.

Reply:
Awesome idea!